contracts/tests/test_validator_set.py

```diff
@@ -142,6 +142,49 @@ def test_cannot_call_finalize_change(validator_set_contract_session, accounts):
         contract.functions.finalizeChange().transact({"from": accounts[0]})
 
 
+def test_finalize_change_update_history_data_correctly(
+    validator_set_contract_session, accounts, web3
```

Comment:
Do I understand correctly that `validator_set_contract_session` carries state between tests? If so I don't think this is a good idea. Why not just make this a single-test scoped fixture?

Reply:
Oh, I wasn't aware about it. To be honest I haven't expected it for such a contract. I removed it at the fixture and all tests are still running.